modules/RTC/JitsiLocalTrack.js

```diff
@@ -134,6 +131,48 @@ JitsiLocalTrack.prototype.isEnded = function () {
 };
 
 /**
+ * Sets handlers to the MediaStreamTrack object that will detect camera issues.
+ */
+JitsiLocalTrack.prototype._initNoDataFromSourceHandlers = function () {
+    if(this.isVideoTrack() && this.videoType === VideoType.CAMERA) {
+        this._setHandler("track_mute", () => {
+            if(this._checkForCameraIssues()) {
+                let now = window.performance.now();
+                this._noDataFromSourceTimout = setTimeout(() => {
+                    this._noDataFromSourceTimout = null;
+                    //unset the handler
+                    this._setHandler("track_unmute", undefined);
+                    if(this._checkForCameraIssues())
+                        this._fireNoDataFromSourceEvent();
+                }, 3000);
+                this._setHandler("track_unmute", () => {
+                    if(this._noDataFromSourceTimout) {
+                        clearTimeout(this._noDataFromSourceTimout);
+                        this._noDataFromSourceTimout = null;
+                        Statistics.sendEventToAll(
+                            this.getType() + ".track_unmute",
+                            window.performance.now() - now);
+                    }
+                });
+            }
+        });
+        this._setHandler("track_ended", () => {
+            if(this._checkForCameraIssues())
+                this._fireNoDataFromSourceEvent();
```

Comment:
If `track_ended` arrives within three seconds of `track_mute`, it looks to me like the above piece of source code will be executed twice: on `track_ended` and when `_noDataFromSourceTimout` fires. I think this can be avoided and the source code can be simplified by having `track_ended` and `setTimeout` execute one and the same function.